pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2604,12 +2631,28 @@ public int getDatabaseMinorVersion() throws SQLException {
     return connection.getServerMinorVersion();
   }
 
-  public int getJDBCMinorVersion() throws SQLException {
-    return 0; // This class implements JDBC 3.0
+  /**
+   * Retrieves the major JDBC version number for this driver.
+   *
+   * @return JDBC version major number
+   */
+  @Override
+  public int getJDBCMajorVersion() {
+    return Integer.parseInt(JDBCVERSION.split("\\.")[0]);
```

Comment:
No, the `mvn.project.property.parsedversion.majorversion` is for the driver version, not for the JDBC specification version, so this should return **4** as the major specification version, this is get from `mvn.project.property.jdbc.specification.version` and parsed the first number.